packages/itwin/tree-widget/src/test/trees/CategoriesVisibilityUtils.test.ts

```diff
@@ -243,4 +251,27 @@ describe("CategoryVisibilityUtils", () => {
       otherViewMock.verify((x) => x.changeSubCategoryDisplay("SubCategoryId", false), moq.Times.never());
     });
   });
+
+  describe("loadCategoriesFromViewport", () => {
+    it("loadCategoriesFromViewport sets subCategories as undefined when subCategories size is 0", async () => {
+      const categoryInfoWithoutSubcategories: Map<Id64String, IModelConnection.Categories.CategoryInfo> = new Map(
+        [[
+          categoryId,
+          {
+            id: categoryId,
+            subCategories: new Map(),
+          },
+        ]]
+      );
+      queryReaderMock.reset();
+      categoriesMock.reset();
+      imodelMock.reset();
```

Comment:
Looks like it should be enough to `reset` and `setup` only `queryReaderMock` and `categoriesMock`

Reply:
fixed